pycqed/analysis_v2/fluxing_analysis.py

```diff
@@ -0,0 +1,212 @@
+            self.run_analysis()
+
+    def extract_data(self):
+        self.raw_data_dict = OrderedDict()
+        a = ma.MeasurementAnalysis(
+            timestamp=self.ts, label=self.label, auto=False)
+        a.get_naming_and_values_2D()
+        a.finish()
+        self.timestamps = [a.timestamp_string]
+        self.raw_data_dict['timestamps'] = self.timestamps
+        self.raw_data_dict['timestamp_string'] = a.timestamp
+        for attr in ['sweep_points', 'sweep_points_2D', 'measured_values',
+                     'parameter_names', 'parameter_units', 'value_names',
+                     'value_units']:
+            self.raw_data_dict[attr] = getattr(a, attr)
+        self.raw_data_dict['folder'] = a.folder
+
+    def process_data(self):
+        self.proc_data_dict = OrderedDict()
+
+        # select the relevant data
+        x = self.raw_data_dict['sweep_points']
+        t = self.raw_data_dict['sweep_points_2D']
+        Z = self.raw_data_dict['measured_values'][self.ch_idx].T
+
+        # fit frequencies to each individual cut (time trace)
+        freqs = []
+        freqs_std = []
+        fit_results = []
+        amps = []
+        for xi, z in zip(x, Z.T):
+            CosModel = lmfit.Model(CosFunc)
+            CosModel.guess = Cos_guess
+            pars = CosModel.guess(CosModel, z, t)
+            fr = CosModel.fit(data=z, t=t, params=pars)
+            amps.append(fr.params['amplitude'].value)
+            freqs.append(fr.params['frequency'].value)
+            freqs_std.append(fr.params['frequency'].stderr)
+            fit_results.append(fr)
+        # N.B. the fit results are not saved in self.fit_res as this would
+        # bloat the datafiles.
+        self.proc_data_dict['fit_results'] = np.array(fit_results)
+        self.proc_data_dict['amp_fits'] = np.array(amps)
+        self.proc_data_dict['freq_fits'] = np.array(freqs)
+        self.proc_data_dict['freq_fits_std'] = np.array(freqs_std)
+
+        # take a Fourier transform (nice for plotting)
+        fft_data = abs(np.fft.fft(Z.T).T)
+        fft_freqs = np.fft.fftfreq(len(t), d=t[1]-t[0])
+        sort_vec = np.argsort(fft_freqs)
+
+        fft_data_sorted = fft_data[sort_vec, :]
+        fft_freqs_sorted = fft_freqs[sort_vec]
+        self.proc_data_dict['fft_data_sorted'] = fft_data_sorted
+        self.proc_data_dict['fft_freqs_sorted'] = fft_freqs_sorted
+
+    def run_fitting(self):
+        super().run_fitting()
+
+        fit_mask = np.where(self.proc_data_dict['amp_fits'] > self.min_fit_amp)
+
+        avoided_crossing_mod = lmfit.Model(avoided_crossing_freq_shift)
+        # hardcoded guesses! Bad practice, needs a proper guess func
+        avoided_crossing_mod.set_param_hint('a', value=3e9)
+        avoided_crossing_mod.set_param_hint('b', value=-2e9)
+        avoided_crossing_mod.set_param_hint('g', value=20e6, min=0)
+        params = avoided_crossing_mod.make_params()
+
+        self.fit_res['avoided_crossing'] = avoided_crossing_mod.fit(
```

Comment:
There is already a general fitting function implemented in the new analysis I dont think that there should be one here

Reply:
The fitting dictionary comes into play here